nixos/modules/services/audio/librespot.nix

```diff
@@ -0,0 +1,205 @@
+        description = "";
+      };
+
+      disable-discovery = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Disable discovery mode";
+      };
+      device = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Audio device to use. (Only with the rodio/portaudio/alsa backends)";
+      };
+      mixer = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Mixer to use (softvol/alsa)";
+      };
+      mixer-name = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "alsa mixer name e.g PCM or Master. Defaults to PCM";
+      };
+      mixer-card = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "alsa mixer card, e.g hw:0 or similar from aplay -l. Defaults to default";
+      };
+      mixer-index = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "alsa mixer index, Index of the cards mixer. Defaults to 0";
+      };
+      inital-volume = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "Initial volume in %, once connected [0-100]";
+      };
+      enable-volume-normalisation = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enables volume normalisation for librespot";
+      };
+      normalisation-pregain = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "A numeric value for pregain (dB). Only used when normalisation active.";
+      };
+      linear-volume = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enables linear volume scaling instead of logarithmic (default) scaling";
+      };
+      disable-gapless = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Disables gapless playback by forcing the sink to close b/w tracks";
+      };
+      zeroconf-port = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "The port that the HTTP server advertised on zeroconf will use [1-65535]. Ports <= 1024 may require root priviledges.";
+      };
+      proxy = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Use a proxy for resolving the access point. Proxy should be an HTTP proxy in the form http://ip:port, and can also be passed using the all-lowercase http_proxy environment variable.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [ librespot ];
+
+    systemd.services.librespot = {
+      description = "librespot";
+      after    = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+      ExecStart = "${pkgs.librespot}/bin/librespot -n ${cfg.name} "
```

Comment:
This would create one long line in `systemd` unit.
You could use [this approach](https://github.com/NixOS/nixpkgs/blob/c83cc9c364649560a7a0dafc610c09cf24e39408/nixos/modules/virtualisation/docker-containers.nix#L224) to create a more readable `systemd` units.